src/shared/watchdog.c

```diff
@@ -114,9 +117,31 @@ int watchdog_set_timeout(usec_t *usec) {
         return r;
 }
 
+usec_t watchdog_runtime_wait(void)
+{
+        usec_t rtwait = USEC_INFINITY;
+
+        if (watchdog_timeout != USEC_INFINITY)
+                rtwait = watchdog_timeout / 2;
+
+        if (rtwait == 0)
+                return 1;
+
+        return rtwait;
```

Comment:
hmm, i think this function should basically calculate `watchdog_last_ping + watchdog_timeout / 2 -  now(…)`, i.e. calculate how much more to sleep at most